lib/hanami/view/helpers/tag_helper.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+
+# Based on ActionView::Helpers::TagHelper, also released under the MIT licence.
+#
+# Copyright (c) David Heinemeier Hansson
+
+require_relative "escape_helper"
+
+module Hanami
+  class View
+    module Helpers
+      module TagHelper
+        require_relative "tag_helper/tag_builder"
+
+        module_function
+
+        # Returns a tag builder for building HTML tag strings.
+        #
+        # @example General usage
+        #   tag.div # => <div></div>
```

Comment:
Mostly because we already had `#html` via `HTMLHelper`, and I was introducing this alongside it just to get your thoughts :) Additionally, it was already called `#tag` in the Rails implementation that I imported.

Now that you're OK with the existing `HTMLHelper` implementation being removed, we have the option to rename it, but I do wonder if `tag` is a better name here, because:

1. Its job is to generate a single tag
2. It's one less character to type!
3. (Least important of all, but still a factor) Matching the name from Rails' helpers might help users understand its usage more readily

For these reasons, my personal inclination would to rename `TagHelper` to `HTMLHelper` (as you requested), but leave the helper method itself named as `#tag`.

I don't feel super strongly about this, though, so I'm keen to hear your thoughts given the above. Then we'll settle on the name.

Reply:
Actually, another reason to leave this named as `TagHelper` is that it also is aware of and supports building non-HTML tags, like SVG tags, demonstrated by the `SVG_SELF_CLOSING_ELEMENTS` constant in `TagBuilder`.

I was reminded of this just as I was cleaning things up in this PR.

So at this point I'd be inclined just to leave everything named as is: `TagHelper#tag`.

Let me know what you think :)